File: q0791/q0791.java
class Solution {
    public String customSortString(String order, String s) {
        Map<Character, Integer> charCount = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            charCount.put(c, charCount.getOrDefault(c, 0) + 1);
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < order.length(); i++) {
            char c = order.charAt(i);
            if (charCount.containsKey(c)) {
                int count = charCount.get(c);
                sb.append(String.valueOf(c).repeat(count));
                charCount.put(c, 0);
            }
        }

        for (char c : charCount.keySet()) {
            int count = charCount.get(c);
            sb.append(String.valueOf(c).repeat(count));
        }
        return sb.toString();
    }
}

File: q0791/q0791.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        # Build the frequency map
        charCount = defaultdict(int)
        for char in s: 
            charCount[char] += 1 

        res = ""
        # Use the character from the order sequence to build the res
        for char in order: 
            if char in charCount:
                res += char * charCount[char]
                charCount[char] = 0
        
        # Use the rest of the characters
        for char, count in charCount.items(): 
            res += char * count
        return res